Sources/CacheAPI-postgres.php

```diff
@@ -108,49 +131,22 @@ public function getData($key, $ttl = null)
 	 */
 	public function putData($key, $value, $ttl = null)
 	{
-		global $db_prefix, $db_connection, $db_persist;
-
-		if (!isset($value))
-			$value = '';
-
-		$ttl = time() + $ttl;
-
-		if (empty($this->pg_put_data_prep))
-		{
-			if (empty($db_persist))
-				$this->pg_put_data_prep = pg_prepare($db_connection, 'smf_cache_put_data',
-					'INSERT INTO ' . $db_prefix . 'cache(key,value,ttl) VALUES($1,$2,$3)
-					ON CONFLICT(key) DO UPDATE SET value = excluded.value, ttl = excluded.ttl'
-				);
-			else
-			{
-				@pg_prepare($db_connection, 'smf_cache_put_data',
-					'INSERT INTO ' . $db_prefix . 'cache(key,value,ttl) VALUES($1,$2,$3)
-					ON CONFLICT(key) DO UPDATE SET value = excluded.value, ttl = excluded.ttl'
-				);
-				$this->pg_put_data_prep = true;
-			}
-		}
+		$ttl = time() + (int) ($ttl !== null ? $ttl : $this->ttl);
 
-		$result = pg_execute($db_connection, 'smf_cache_put_data', array($key, $value, $ttl));
-
-		if (pg_affected_rows($result) > 0)
-			return true;
+		if ($value === null)
+			$result = pg_execute($this->db_connection, 'smf_cache_delete_data', array($key));
```

Comment:
The SMF cache system is special in that it expects a null value to trigger a delete.

Reply:
Where is that used?
And why is not a negative ttl value used?

Reply:
To make my point clear here,
the delete looks fine,
but here we got two issue in my eyes:
This behavior is not descripte here https://github.com/SimpleMachines/SMF2.1/blob/73cee9a121eb1047518f4fa119588d893623dc11/Sources/Class-CacheAPI.php#L83-L92
Secondly i would expect that this behavior realized by a negativ ttl instead of a null value.

about null values vs  negative ttl ->
null value we use in smf for example in ip address logic so maybe i want to save this in a cache i can't when the null value always get deleted.
negative ttl is never a value from "common" smf it's special command from cache logic.